tests/simulator/test_restarts.py

```diff
@@ -97,6 +97,59 @@ def ddt_test_id(self):
         return self._testMethodName.split("_")[-1]
 
 
+    def check_diags(self, diag_dir0, diag_dir1, pops, timestamps):
+
+        def check(qty0, qty1, checker):
+            checks = 0
+            for ilvl, lvl0 in qty0.patch_levels.items():
+                patch_level1 = qty1.patch_levels[ilvl]
+                for p_idx, patch0 in enumerate(lvl0):
+                    patch1 = patch_level1.patches[p_idx]
+                    for pd_key, pd0 in patch0.patch_datas.items():
+                        pd1 = patch1.patch_datas[pd_key]
+                        self.assertNotEqual(id(pd0), id(pd1))
+                        checker(pd0, pd1)
+                        checks += 1
+            return checks
+
+        def check_particles(qty0, qty1):
+            return check(qty0, qty1, lambda pd0, pd1: self.assertEqual(pd0.dataset, pd1.dataset))
+
+        def check_field(qty0, qty1):
+            return  check(qty0, qty1, lambda pd0, pd1: np.testing.assert_equal(pd0.dataset[:], pd1.dataset[:]))
+
+
+        def count_levels_and_patches(qty):
+            n_levels = len(qty.patch_levels)
+            n_patches = 0
+            for ilvl, lvl in qty.patch_levels.items():
+                n_patches += len(qty.patch_levels[ilvl].patches)
+            return n_levels, n_patches
+
+        n_quantities_per_patch = 20
```

Comment:
do you think we can get that magic number from somewhere so it does not become inconsistent later?
Could be a `run.GetAllAvailableQties(time=0)` that returns the list of names of all quantities (exploding vectors into components) and `n_quantities_per_patch=20` would become 
`n_quantities_per_patch = len(run.GetAllAvailableQties())`
(just an example there might be a better idea)

Reply:
the following checks have to stay consistent if that total changes though...

Reply:
number of quantities is configured in the diagnostic setup for this test, it's not for "restarts" per say

Reply:
it has a name, so it's not really a magic number, but I see you point, maybe it should be in `tests/diagnostics/__init__.py` as the number of quantities in the diagnostic output is a product of the "dump_all_diags" function